crates/ruff/tests/format.rs

```diff
@@ -90,6 +90,172 @@ fn format_warn_stdin_filename_with_files() {
     "###);
 }
 
+#[test]
+fn nonexistent_config_file() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(["format", "--config", "foo.toml", "."]), @r###"
+        success: false
+        exit_code: 2
+        ----- stdout -----
+
+        ----- stderr -----
+        error: invalid value 'foo.toml' for '--config <CONFIG_OPTION>'
+
+          tip: The `--config` flag must either be a path to a `.toml` configuration file or a TOML `<KEY> = <VALUE>` pair overriding a specific config setting
```

Comment:
I would say "configuration setting" instead of "config setting" for consistency with "configuration file".

I also have a minor preference for "option" over "setting", not sure if we're consistent about that terminology in general.

Reply:
You may be able to drop "TOML" from this as well

Reply:
I've kept "TOML" in here for now, as otherwise I think it's a little confusing in some of the error messages when we immediately start talking about a "TOML parse error" in the following lines